File: 2023/06/04/02_menghitung_karakter.py
"""
Buat program Python yang meminta pengguna untuk memasukkan sebuah string dan kemudian menampilkan jumlah karakter dalam string tersebut.
"""


# validasi input
def validasi_input(input_string):
    """
    Function ini digunakan untuk memvalidasi input dari pengguna.
    """
    while True:
        user_input = input(input_string)
        if len(user_input) == 0:
            print("Input tidak boleh kosong. Silakan masukkan kembali.")
        else:
            return user_input


# menghitung jumlah karakter dalam string
def hitung_karakter(string):
    """
    Function ini digunakan untuk menghitung jumlah karakter dalam string.
    """
    return len(string)


def main():
    # meminta input dari pengguna
    input_1 = validasi_input("Masukkan sebuah string: \n")

    # menghitung jumlah karakter dalam string
    jumlah_karakter = hitung_karakter(input_1)

    # menampilkan hasil
    print(f"Jumlah karakter dalam string {input_1} adalah {jumlah_karakter}")


if __name__ == "__main__":
    main()

File: 2023/06/05/02_menghitung_karakter.c
// Buat program C yang meminta pengguna untuk memasukkan sebuah string dan kemudian menampilkan jumlah karakter dalam string tersebut.

#include <stdio.h>

// menghitung karakter
int hitungKarakter(char *str)
{
    int i = 0;
    while (str[i] != '\0')
    {
        i++;
    }
    return i;
}

// program utama
int main()
{
    // deklarasi variabel
    char str[100];

    // input
    printf("Masukkan sebuah string: ");
    scanf("%s", str);

    // hitung jumlah karakter
    int jumlahKarakter = hitungKarakter(str);

    // output
    printf("Jumlah karakter: %d\n", jumlahKarakter);

    // opsi mengulang atau keluar
    char opsi;
    printf("Ulangi? (y/n) ");
    scanf(" %c", &opsi);
    if (opsi == 'y' || opsi == 'Y')
    {
        main();
    }
    else
    {
        printf("Terima kasih.\n");
        return 0;
    }

    return 0;
}

// catatan: untuk menghitung jumlah karakter dalam sebuah string, kita dapat menggunakan fungsi strlen() yang terdapat dalam library string.h. Namun, untuk memahami konsep perulangan, kita akan membuat fungsi sendiri untuk menghitung jumlah karakter dalam sebuah string.